mitodl/mitxpro
# test(emeritus_api): add more tests for emeritus API ingestion

### What are the relevant tickets?




https://github.com/mitodl/mitxpro/issues/3028

### Description (What does it do?)

Adds more tests for the external course sync to make it more robust. Also, makes some refactoring to show filtered stats and make the object creation more consistent.

### How can this be tested?


- Checks should pass
- Verify that all tests mentioned in the related issue are added.
- Verify that it covers the edge cases around draft and live states of Certificate pages and Course Pages.
- Verify that the stats logged by the command `./manage.py sync_external_course_runs --vendor emeritus` are cleaner and much more accurate.
- Test the ExternalCoursePage and CertificatePage states. Command should not update the status while updating the data.
  - Run the command on master.
  - Make a few changes in a live course and save as draft, this state is called `live + draft`
  - Unpublish a live course.
  - Make changes to a live course and publish it.
  - Run the command on this branch and it should update the data but not the status.
- Test the above cases for the CertificatePage.
